include/cuco/detail/__config

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#if defined(CUDART_VERSION) && (CUDART_VERSION >= 11000) && defined(__CUDA_ARCH__) && \
+  (__CUDA_ARCH__ >= 700)
+#define CUCO_HAS_CUDA_BARRIER
+#endif
+
+#if defined(CUDART_VERSION) && (CUDART_VERSION >= 11100)
+#define CUCO_HAS_CG_MEMCPY_ASYNC
+#endif
+
+#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 700)
+#define CUCO_HAS_LARGE_TYPE_SUPPORT
```

Comment:
I think a more appropriate name would be `CUCO_HAS_VOLTA_THREAD_MODEL` or `CUCO_HAS_INDEPENDENT_THREADS`.

The "large type support" is a consequence of needing to use locking, which needs the independent thread model.

Reply:
This reminds me that in order to be friendly to nvc++, we should probably be switching to use `NV_IF_TARGET`. That'll be a different issue though.